Allow key-value options to be passed in for harmonization flows in UI

wooldridge opened the pull request:
The features allows the user to add a configurable number of key-value options.
The key-value list is built as a configurable Angular component. 
Harmonization flow options in the UI are persisted in local storage. 
Also fixes #684 .

marklogic-builder commented:
Can one of the admins verify this patch?

marklogic-builder commented:
Can one of the admins verify this patch?

marklogic-builder commented:
Jenkins Build successful 
http://jenkins.marklogic.com/job/Labs/job/DHF_b9_nightly_linux_cluster/9
 509 tests run, 0 skipped, 0 failed.

wooldridge commented:
@aebadirad, OK, looks like this got inadvertently closed when I was doing housekeeping during the rebase of my branch (in the process, I deleted the wooldridge:DHFPROD-167 branch), my bad. I can refile a PR with the updated, rebased branch.

wooldridge commented:
Opened this feature again as: https://github.com/marklogic-community/marklogic-data-hub/pull/713
